test/parallel_api/numeric/numeric.ops/reduce_by_segment.pass.cpp

```diff
@@ -368,16 +350,13 @@ main()
     test_flag_pred<sycl::usm::alloc::device, class KernelName2, dpl::complex<float>>();
 #endif // TEST_DPCPP_BACKEND_PRESENT
 
-    run_test<::std::uint64_t, UserBinaryPredicate<::std::uint64_t>, MaxFunctor<::std::uint64_t>>();
+    run_test<::std::uint32_t, UserBinaryPredicate<::std::uint32_t>, MaxFunctor<::std::uint32_t>>();
     run_test<::std::complex<float>, UserBinaryPredicate<::std::complex<float>>, MaxFunctor<::std::complex<float>>>();
 
     run_test<int, ::std::equal_to<int>, ::std::plus<int>>();
-    run_test<float, ::std::equal_to<float>, ::std::plus<float>>();
```

Comment:
@mmichel11, you exclude test for `float` type: but float type available on each device, and `double` type only on some devices.
So why do you think that this approach is correct?

Reply:
That is a good point. I could add back the `float` test and remove the `double`. I don't think there is much of a need to test both `float` and `double` with the same operator and predicate. If one passes, I suspect the other one will likely be fine as well.